lib/hedera/HederaNetwork.ts

```diff
@@ -143,4 +88,67 @@ export class HederaNetwork {
     }
     return networkInfo;
   }
+
+  public readonly nodes: HederaNodesAddressBook;
+  public readonly mirror: HederaRestMirror;
+
+  public constructor(private readonly params: HederaNetworkConstructorArgs) {
+    if (typeof params.nodes === "string") {
+      this.nodes = HederaNetwork.parseNetworkAddressBookFrom(params.nodes);
+    } else {
+      this.nodes = params.nodes;
+    }
+    if (!ACCEPTED_NETWORK_NAMES.includes(this.name)) {
+      throw new EnvironmentInvalidError(
+        `There is no such '${
+          this.params.name
+        }' network available. In order to continue, please choose a valid name from: ${ACCEPTED_NETWORK_NAMES.join(
+          ", "
+        )}`
+      );
+    }
+
+    this.mirror = new HederaRestMirror(
```

Comment:
We are aware of some apps who are not using the Hedera offered mirror nodes on public networks. Would we be able to accommodate custom addresses for public networks as well?